Zend/tests/immutable/properties/property_magic_set2.phpt

```diff
@@ -0,0 +1,47 @@
+--TEST--
+Test that the __set() magic method works as expected for immutable classes
+--FILE--
+<?php
+
+immutable class Foo
+{
+    private array $properties = [
+        "property1" => "foo",
+        "property2" => null,
+        "property3" => [],
+    ];
+
+    public function __set($name, $value): void
+    {
+        $this->properties[$name] = $value;
```

Comment:
As arrays are by-value, isn't this violating the immutability contract by modifying the array?

Reply:
Yes, it is, I only haven't implemented this restriction yet (the test should have been XFAIL).